src/main/java/org/jenkinsci/plugins/workflow/steps/UnstableStep.java

```diff
@@ -72,9 +83,12 @@ private UnstableStepExecution(UnstableStep step, StepContext context) {
 
         @Override
         protected Void run() throws Exception {
-            getContext().get(FlowNode.class).addOrReplaceAction(new WarningAction(Result.UNSTABLE).withMessage(step.message));
-            getContext().get(Run.class).setResult(Result.UNSTABLE);
-            getContext().get(TaskListener.class).getLogger().append("WARNING: ").println(step.message);
+            StepContext context = getContext();
+			context.get(FlowNode.class).addOrReplaceAction(new WarningAction(Result.UNSTABLE).withMessage(step.message));
+			if (step.markBuildResult) {
+				context.get(Run.class).setResult(Result.UNSTABLE);
+			}
```

Comment:
tabs

Reply:
sorry, new eclipse so preferences are gone...

Reply:
maybe it's the case to add a checkstyle rule ;)

Reply:
fixed